src/TestCase.ts

```diff
@@ -0,0 +1,173 @@
+import * as yaml from "js-yaml";
+import * as vscode from "vscode";
+import { Position, Range, Selection } from "vscode";
+import NavigationMap from "./NavigationMap";
+import { ActionType, PartialTarget, PrimitiveTarget, Target } from "./Types";
+
+export type SerializedPosition = {
+  line: number;
+  character: number;
+};
+
+export type SerializedRange = {
+  start: SerializedPosition;
+  end: SerializedPosition;
+};
+
+export type SerializedSelection = {
+  anchor: SerializedPosition;
+  active: SerializedPosition;
+};
+
+export function serializeRange(range: Range): SerializedRange {
+  return {
+    start: serializePosition(range.start),
+    end: serializePosition(range.end),
+  };
+}
+
+export function serializeSelection(selection: Selection): SerializedSelection {
+  return {
+    active: serializePosition(selection.active),
+    anchor: serializePosition(selection.anchor),
+  };
+}
+
+export function serializePosition(position: Position): SerializedPosition {
+  return { line: position.line, character: position.character };
+}
+
+type Command = {
+  actionName: ActionType;
+  partialTargets: PartialTarget[];
```

Comment:
For completeness I'd be tempted to also capture the full target, even if we don't end up using it for anything